src/cli/build.rs

```diff
@@ -16,14 +25,201 @@ pub struct BuildCommand {
+        }
+
+        // Create the workdir ------------------------------------------------------------
+        println!("Creating workdir {}", workdir);
+        std::fs::create_dir_all(&workdir)?;
+
+        // If the kernel is a directory, copy it into the workdir
+        if std::fs::metadata(&self.kernel).is_ok() {
+            copy_dir(&self.kernel, workdir)?;
+        } else {
+            // If the kernel doesn't exist, display error message
+            println!("Kernel doesn't exist.");
+            return Ok(());
+        }
+
+        // If the rootfs is a directory, copy it into the workdir
+        if std::fs::metadata(&self.rootfs).is_ok() {
+            copy_dir(&self.rootfs, workdir)?;
+        } else {
+            // If the rootfs doesn't exist, display error message
+            println!("Rootfs doesn't exist.");
+            return Ok(());
+        }
+
+        // Get the rootfs filename
+        let rootfs_path = std::path::Path::new(&self.rootfs);
+        let rootfs_filename = rootfs_path.file_name().unwrap().to_str().unwrap();
+
+        // If kaps exists, copy it into the workdir rootfs
+        if std::fs::metadata(&self.kaps).is_ok() {
+            println!(
+                "Copying kaps to workdir rootfs {}{}",
+                workdir, rootfs_filename
+            );
+
+            std::fs::copy(&self.kaps, &format!("{}/{}/kaps", workdir, rootfs_filename))?;
+        } else {
+            // If kaps doesn't exist, display error message
+            println!("Kaps doesn't exist.");
+            return Ok(());
+        }
+
+        // If bundle exists, copy  the directory into the workdir rootfs
+        if std::fs::metadata(&self.bundle).is_ok() {
+            copy_dir(&self.bundle, &format!("{}{}", workdir, rootfs_filename))?;
+        } else if self.bundle.contains("http") {
+            // If bundle is a url, download it
+            /* TODO: implement */
+            // display error message
+            println!("Bundle is a url, not implemented yet.");
+            return Ok(());
+        } else {
+            // If bundle doesn't exist, display error message
+            println!("Bundle doesn't exist.");
+            return Ok(());
+        }
+
+        // Configure rootfs ------------------------------------------------------------
+        println!("Configuring rootfs");
+
+        // Change the init script to use the kaps bundle
+        println!("Changing init script to use the kaps bundle");
+        let init_script = format!("{}/{}/init", workdir, rootfs_filename);
+        // Remove the old init script
+        std::fs::remove_file(init_script.clone()).unwrap();
+
+        // Write the new init script
+        let mut init_script_file = std::fs::File::create(init_script.clone())?;
+        println!("Writing new init script");
+        let kaps_name = self.kaps.split('/').last().unwrap();
+        let bundle_name = self.bundle.split('/').last().unwrap();
```

Comment:
Lots of unwraps, lots of potential panics. Return an error (through e.g.`map_err`).